docs/crawl_data_parsing.md

```diff
@@ -0,0 +1,97 @@
+# Crawl Data Parsing
+
+As mentioned in the [project README](../README.md), one thing that sets this application apart from just a web crawl is the parsing of metadata records that represent the websites crawled.  This is the work of the `harvester.parse.CrawlMetadataParser` class.
+
+## What crawled sites are included as metadata records?
+
+A browsertrix web crawl generates a bunch of files as the result of the crawl:
+
+```text
+├── archive
+│   ├── rec-20230929152646786455-3e0077b1f009.warc.gz
+│   ├── rec-20230929152647253458-3e0077b1f009.warc.gz
+│   ├── rec-20230929152648227041-3e0077b1f009.warc.gz
+│   ├── rec-20230929152648249931-3e0077b1f009.warc.gz
+│   ├── rec-20230929152648436859-3e0077b1f009.warc.gz
+│   ├── rec-20230929152648567703-3e0077b1f009.warc.gz
+│   ├── rec-20230929152648603443-3e0077b1f009.warc.gz
+│   ├── rec-20230929152648639794-3e0077b1f009.warc.gz
+│   ├── rec-20230929152649186320-3e0077b1f009.warc.gz
+│   └── rec-20230929152649791199-3e0077b1f009.warc.gz
+├── homepage.wacz
+├── indexes
+│   └── index.cdxj
+├── logs
+│   └── crawl-20230929152634980.log
+├── pages
+│   └── pages.jsonl
+├── static
+└── templates
+```
+
+Note the [WACZ](https://replayweb.page/docs/wacz-format) file here, `homepage.wacz` (where "homepage" is an arbitrary `--crawl-name`).  This file is a compressed form of the entire crawl, and is the only asset parsed by this application when generating metadata records.  This is handy for multiple reasons:
+  * this file can be saved or copied, representing the crawl in its totality
+  * parsing data from the crawl is only concerned with a single file as the entrypoint (though multiple files are later accessed inside) allowing creation of metadata records even from a remote file in S3
+
+The structure of the WACZ file is nearly identical to the uncompressed crawl results, with some minor, but important, additions:
+```text
+.
+├── archive
```

Comment:
To make sure I understand, everything from lines 39-59 reflect the contents of `homepage.wacz` when it is decompressed?

Reply:
That's correct!  It's just a zip file of the full results (plus a couple extra files) of a crawl.